x-pack/plugin/xstateless/build.gradle

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+apply plugin: 'elasticsearch.internal-es-plugin'
+apply plugin: 'elasticsearch.internal-cluster-test'
+apply plugin: 'elasticsearch.internal-test-artifact'
+esplugin {
+  name = 'xstateless'
```

Comment:
note. We need a different name than `stateless` to differentiate this new plugin in the public repository. Especially a different name for the folder under `x-pack`. Else the same name `stateless` would create difficulties to compile and have things working in serverless. Seems like current gradle/kotlin expects plugins to have unique names, if I'm not mistaken.